src/compiler/checker.ts

```diff
@@ -6685,6 +6688,23 @@ namespace ts {
         }
 
         function resolveTypeReferenceMembers(type: TypeReference): void {
+
+            if (type.target === globalThisType && type.typeArguments && type.typeArguments[0].flags & TypeFlags.TypeParameter &&

```

Comment:
I also don't understand what `globalThisType` is doing here (unless I'm completely mistaken and it's a marker type rather than the type of `globalThis`.)

Reply:
It's the `ThisType<T>` marker type.